aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -614,12 +612,17 @@ module aptos_framework::object {
         };
     }
 
-    #[deprecated]
-    /// Previously allowed to burn objects, has now been disabled.  Objects can still be unburnt.
-    ///
-    /// Please use the test only [`object::burn_object`] for testing with previously burned objects.
-    public entry fun burn<T: key>(_owner: &signer, _object: Object<T>) {
-        abort error::permission_denied(EBURN_NOT_ALLOWED)
+    /// Add a TombStone to the object.  The object will then be interpreted as hidden via indexers.
+    /// This only works for objects directly owned and for simplicity does not apply to indirectly owned objects.
+    /// Original owners can reclaim burnt objects any time in the future by calling unburn.
+    /// Please use the test only [`object::burn_object_with_transfer`] for testing with previously burned objects.
+    public entry fun burn<T: key>(owner: &signer, object: Object<T>) acquires ObjectCore {
+        let original_owner = signer::address_of(owner);
```

Comment:
recommend asserting that it isn't already burned, otherwise we can cause the user to keep calling this and potentially wasting gas